dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java

```diff
@@ -918,9 +907,18 @@ private OrganisationUnit getOrganisationUnit( IdSchemes idSchemes, String id )
             () -> manager.getObject( OrganisationUnit.class, idSchemes.getOrgUnitIdScheme(), id ) );
     }
 
-    private DataElement getDataElement( IdScheme idScheme, String id )
+    /**
+     * Get DataElement by given uid
+     *
+     * @return FALSE if currentUser doesn't have READ access to given
+     *         DataElement OR no DataElement with given uid exist TRUE if
+     *         DataElement exist and currentUser has READ access
+     */
+    private boolean getDataElement( String userUid, String dataElementUid )
```

Comment:
I think the name of this method and the name of the dataElementCache isn't fully descriptive anymore, as it represents whether the current user has access to the data element, and not the data element itself.

This also begs for a better and more holistic solution for checking a user's access to a metadata object in memory, but that is for another day.